### PR TITLE
Add callback for call and hangup on a contact quiddity

### PR DESCRIPTION
**Pull request by francoisph**

### CODE REVIEW

File: client/src/view/scenic/pages/matrix/contacts/ContactDestinationView.js [hunk 1]
```diff
@@ -60,14 +64,21 @@ define( [
 
         callDestination: function() {
             this.closeMenu();
-            this.model.set( 'actions', { 'call': false, 'hangup': true , 'remove': true } );
-            this.model.call();
+            this.model.call(( error ) =>  {
+                if( !error ) {
+                    this.model.set( 'actions', { 'call': false, 'hangup': true , 'remove': true } );
```
Comment (ubald) [inline client/src/view/scenic/pages/matrix/contacts/ContactDestinationView.js]: The model could do this on its own

File: client/src/view/scenic/pages/matrix/contacts/ContactDestinationView.js [hunk 2]
```diff
@@ -60,14 +64,21 @@ define( [
...
+            
         },
 
         hangDestination: function() {
             this.closeMenu();
-            this.model.set( 'actions', { 'call': true, 'hangup': false , 'remove': true } );
-            this.model.hangUp();
+            this.model.hangUp(( error ) =>  {
+                if( !error ) {
+                    this.model.set( 'actions', { 'call': true, 'hangup': false , 'remove': true } );
```
Comment (ubald) [inline client/src/view/scenic/pages/matrix/contacts/ContactDestinationView.js]: The model could do this on its own

File: client/src/view/scenic/pages/matrix/contacts/ContactDestinationView.js [hunk 3]
```diff
@@ -48,6 +48,10 @@ define( [
          */
         initialize: function( ) {
             DestinationView.prototype.initialize.apply(this, arguments);
+
+            // set the actions  
+            var calling = this.model.get('send_status') == 'calling' ? true : false;
+            this.model.set( 'actions', { 'call': !calling, 'hangup': calling , 'remove': true } );
```
Comment (ubald) [inline client/src/view/scenic/pages/matrix/contacts/ContactDestinationView.js]: The model could do this on its own

**Comment by francoisph** [comment]
@ubald code review updated

File: client/src/model/sip/Contact.js [hunk 4]
```diff
@@ -55,34 +55,41 @@ define( [
...
-                    self.scenic.sessionChannel.vent.trigger( 'error', error );
-                }
-                if ( cb ) {
-                    cb( error );
-                }
-            } );
+
+            if ( this.get('connections').length > 0 ) {
+                this.scenic.socket.emit( 'sip.contact.call', this.id, ( error, called ) => {
+                    console.log('eror', error)
```
Comment (ubald) [inline client/src/model/sip/Contact.js]: Remove `console.log`

File: client/src/model/sip/Contact.js [hunk 5]
```diff
@@ -55,34 +55,41 @@ define( [
...
+                    console.log('called', called)
```
Comment (ubald) [inline client/src/model/sip/Contact.js]: Remove `console.log`

File: client/src/model/sip/Contact.js [hunk 6]
```diff
@@ -55,34 +55,41 @@ define( [
...
-            this.scenic.socket.emit( 'sip.contact.call', this.id, function ( error ) {
-                if ( error ) {
-                    self.scenic.sessionChannel.vent.trigger( 'error', error );
-                }
-                if ( cb ) {
-                    cb( error );
-                }
-            } );
+
+            if ( this.get('connections').length > 0 ) {
```
Comment (ubald) [inline client/src/model/sip/Contact.js]: That wasn't there before... what does it add?

**Pull request merged**